php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -1227,16 +1227,18 @@ class NDB_BVL_Instrument extends NDB_Page
     function XINRunElementRules($elname, $elements, $rules)
     {
         $errors = array();
+        //Conditions for the rule to be true, thus the element to be required.
+        $rule_outcomes = array();
 
         foreach ($rules['rules'] AS $rule) {
-            //Conditions for the rule to be true, thus the element to be required.
-            $rule_outcomes = array();
             //Loop through the assigned rules (which is the array of formatted
             //statements passed in XINRegisterRule)
             //If this is an OR rule using two different controllers explode it
             //at the pipe.  ex: q_1{@}=={@}yes|q_2{@}=={@}yes
-            if (stristr(substr($rule, strpos($rule, "|")), "{@}")) {
-                $rules_array = explode("|", $rule);
+            $isInterfieldOr = false;
+            if (substr_count($rule, '{@}') > 2 && strstr($rule, '|')) {
+                $rules_array    = explode("|", $rule);
```

Comment:
don't think i've ever encountered anything like that. but if someone is using this sort of rule, then it would have never worked in the first place. i think xin rules should just support the basics and anything more advanced than that should warrant a custom function to be coded in the instrument itself